crates/ty_python_semantic/src/types/call/arguments.rs

```diff
@@ -178,26 +148,31 @@ impl<'a, 'db> CallArgumentTypes<'a, 'db> {
             // Find the next type that can be expanded.
             let expanded_types = loop {
                 let arg_type = self.types.get(index)?;
-                if let Some(expanded_types) = expand_type(db, *arg_type) {
-                    break expanded_types;
+                if let Some(arg_type) = arg_type {
+                    if let Some(expanded_types) = expand_type(db, *arg_type) {
+                        break expanded_types;
+                    }
                 }
                 index += 1;
             };
 
-            let mut expanded_arg_types = Vec::with_capacity(expanded_types.len() * previous.len());
+            let mut expanded_arguments = Vec::with_capacity(expanded_types.len() * previous.len());
 
             for pre_expanded_types in previous.iter() {
                 for subtype in &expanded_types {
-                    let mut new_expanded_types = pre_expanded_types.clone();
-                    new_expanded_types[index] = *subtype;
-                    expanded_arg_types.push(new_expanded_types);
+                    let mut new_expanded_types = pre_expanded_types.to_vec();
+                    new_expanded_types[index] = Some(*subtype);
+                    expanded_arguments.push(CallArguments {
+                        arguments: self.arguments.clone(),
+                        types: new_expanded_types,
+                    });
```

Comment:
Would it make sense to use a constructor function on `CallArguments` instead of initializing the fields directly so that we can assert that `argumetns` and `types` have the same len